### PR TITLE
Allow addon menu items to run arbitrary code on click

### PR DESCRIPTION
**Pull request by troggy**
This will allow viewless addons or addons having separate view, not only tab view in home contoller.

[Example](https://github.com/troggy/copay-colored-coins-plugin/commit/179cc7b4f0365743ad97a5f6e66c092d91a9e018#diff-ce1d47e10e38c9c3cede9d6c803b88d3R24)

### CODE REVIEW

File: src/js/controllers/index.js [hunk 1]
```diff
@@ -140,27 +153,30 @@ angular.module('copayApp.controllers').controller('indexController', function($r
...
-    if (!self.tab)
+    if (!self.tab || !$state.is('walletHome'))
       self.tab = 'walletHome';
 
-    if (document.getElementById(self.tab)) {
-      document.getElementById(self.tab).className = 'tab-out tab-view ' + self.tab;
-      var old = document.getElementById('menu-' + self.tab);
-      if (old) {
-        old.className = '';
+    go.path('walletHome', function() {
```
Comment (matiu) [inline src/js/controllers/index.js]: Why you removed go.path()? IIUC, go.path('walletHome', is needed for setting the state to `walletHome` when / if set path is called from outside a inner tab.

Comment (troggy) [inline src/js/controllers/index.js]: You probably wanted to ask why I've **added** go.path() ? Exactly for that purpose -- to set state to ``walletHome`` when ``setTab`` is called from other state.

**Comment by troggy** [comment]
Rebased

**Comment by troggy** [comment]
Rebased

**Comment by cmgustavo** [comment]
LGTM

**Comment by cmgustavo** [comment]
Need rebase

**Comment by troggy** [comment]
Rebased

**Comment by cmgustavo** [comment]
ACK

**Comment by cmgustavo** [comment]
I found an issue in this PR using `bitcoin:` URI. 

* When Copay is opened by a bitcoin URI, and after choosing the folder, it finally goes to Home (instead Send).
* Also if you are testing in a browser, refresh the page in, for example, `Preferences`. After page was loaded, it returns to Home.

@troggy: let me know if you have the same trouble.

**Comment by troggy** [comment]
@cmgustavo Yes, I have the same. The reason for this is that go.walletHome (which calls setTab) is being called even for Preferences - why it should?

**Comment by troggy** [comment]
Without my code it just tries to setTab several times and eventually fails. For user everything is fine, but this kind of logic is very confusing for developer

**Comment by troggy** [comment]
My reasoning is following: ``setTab`` is a function to set tab on the wallet home - it is the only view that has tabs. So in theory it should be perfectly safe to switch to Home in setTab - we shouldn't be calling setTab from other views anyway. But in reality it's flawed - ``go.walletHome()`` is being called for ANY view in ``Local/NewFocusedWallet`` event handler.

**Comment by troggy** [comment]
@cmgustavo Please take a look at https://github.com/bitpay/copay/pull/3153 if it resolves the issue for you?

**Pull request merged**